File: Clase 06-02-2016/Laboratorio Eder Gonzalez/Laboratorio Eder Gonzalez/LlamadaLocal.vb
Public Class LlamadaLocal
    Inherits Llamada

    Private _Precio As Double = 1.0

    Public Property Precio As Double
        Get
            Return _Precio
        End Get
        Set(value As Double)
            _Precio = value
        End Set
    End Property

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(ByVal NumeroOrigen As String, ByVal NumeroDestino As String, ByVal Duracion As Double, ByVal Precio As Double)
        MyBase.New(NumeroOrigen, NumeroDestino, Duracion)
        Me.Precio = Precio
    End Sub

    Public Overrides Function CalcularPrecio() As Double
        Dim Resultado As Double = 0.00
        Resultado = Precio * Duracion
        Return Resultado
    End Function

    Public Overrides Function tostring() As String
        Dim Resultado As String = String.Empty
        Resultado = "Se registro una llamada local " & vbCrLf &
            " Numero Origen: " & NumeroOrigen & vbCrLf &
            " Numero Destino: " & NumeroDestino & vbCrLf &
            " Costo: " & CalcularPrecio() & vbCrLf &
            " Duracion: " & Duracion

        Return Resultado
    End Function

End Class

File: CentralitaC/CentralitaC/LlamadaLocal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CentralitaC
{
    class LlamadaLocal : Llamada
    {
        private double Precio { get; set; } = 1.0;
        
        public LlamadaLocal() { new Llamada(); }

        public LlamadaLocal(string NumeroOrigen, string NumeroDestino, double Duracion, double Precio)
        {
            new Llamada(NumeroOrigen, NumeroDestino, Duracion);
            this.Precio = Precio
        }

        public override double CalcularPrecio()
        {
            Double Resultado = 0.00;
            Resultado = Precio * base.Duracion;
            return Resultado;
        }

        public override string ToString()
        {
            string Resultado = string.Empty;

            Resultado = "Se registro una llamada local " +
                " Numero Origen: "+NumeroOrigen+
                " Numero Destino: "+NumeroDestino+
                " Costo: " + CalcularPrecio() +
                " Duracion: " + Duracion ;

            return Resultado;
        }
    }
}
